verticapy/toolbox.py

```diff
@@ -444,16 +564,32 @@ def get_session(add_username: bool = True):
 
 
 # ---#
-def gen_tmp_name(schema: str = "", name: str = ""):
-    session_user = get_session()
-    L = session_user.split("_")
-    L[0] = "".join(filter(str.isalnum, L[0]))
-    L[1] = "".join(filter(str.isalnum, L[1]))
-    random_int = random.randint(0, 10e9)
-    name = '"_verticapy_tmp_{}_{}_{}_{}_"'.format(name.lower(), L[0], L[1], random_int)
-    if schema:
-        name = "{}.{}".format(quote_ident(schema), name)
-    return name
+def get_vertica_type(dtype):
+    if dtype in (str, "str", "string"):
+        dtype = "varchar"
+    elif dtype == float:
+        dtype = "float"
+    elif dtype == int:
+        dtype = "integer"
+    elif dtype == datetime.datetime:
+        dtype = "datetime"
+    elif dtype == datetime.date:
+        dtype = "date"
+    elif dtype == datetime.time:
+        dtype = "time"
+    elif dtype == datetime.timedelta:
+        dtype = "interval"
+    elif dtype == datetime.timezone:
+        dtype = "timestamptz"
+    elif dtype in (np.ndarray, np.array, list):
+        dtype = "array"
+    elif dtype == dict:
+        dtype = "row"
+    elif dtype == tuple:
+        dtype = "set"
+    elif isinstance(dtype, str):
+        dtype = dtype.lower()
```

Comment:
Wouldn't it be better to error-out when the input `dtype` doesn't satisfy any condition?

Reply:
Not really because the input can already be a Vertica dtype.